### PR TITLE
feat(@schematics/angular): add generator for interceptor

### PR DESCRIPTION
**Pull request by FG-33**
It is now possible to generate interceptors using the angular-cli (see #6937):

`$ ng g interceptor(ic) test`

Result:
```
CREATE src/app/test.interceptor.spec.ts (417 bytes)
CREATE src/app/test.interceptor.ts (419 bytes)
```

Is master the correct one to merge into? The documentation says devkit:master but there's no such branch. devkit is too far behind I guess.

### CODE REVIEW

**Comment by FG-33** [comment]
@googlebot I fixed it.

**Comment by vikerman** [comment]
Thanks for the PR!

We think both the frequency of adding an interceptor and the complexity of the adding one are pretty low. So we are leaning towards not needing this schematic for now.

**Comment by FG-33** [comment]
> Thanks for the PR!
> 
> We think both the frequency of adding an interceptor and the complexity of the adding one are pretty low. So we are leaning towards not needing this schematic for now.

If you say so. Thanks for the feedback though.

**Comment by crutchcorn** [comment]
While in the end of any project, the maintainers are the final deciding voice and that within itself is something I respect greatly, it seems to me (from a user looking at the project) that many of the schematics are simply convinience tools. 

EG, the component generator. The component complexity seems relatively low, especially once you have an initial component to copy+paste off of. However, for an interceptor I don't often use them for various projects and then have to Google the correct usage of it (or find which of my many Angular projects has one to copy off of), so having this generator (even if the complexity is low overall) is a big convinience for me as I end up using it _enough_ to come looking for this issue every now and then 

The engagement with the linked issue (which has unfortunately fallen victim to the "+1" comment wave, but even with reacts alone) seems to show a further community interest 

Just giving my own thoughts, huge thank you to everyone on the Angular team for your persistent hard-work on the project regardless of where this issue lands

**Comment by alan-agius4** [comment]
Can you also add `interceptor` in https://github.com/angular/angular-cli/blob/ace02f6c2a121a9120fd368617fe3f70992b47f6/packages/schematics/angular/application/index.ts#L172

**Comment by FG-33** [comment]
@clydin I addressed all the mentioned changes. If anything's missing, let me know.

**Comment by simonua** [comment]
I am glad this got merged in. Echoing @crutchcorn's appreciation for the Angular team as well as his reasoning for having a generator, we have several teams in my organization that are starting to develop with Angular 8 now and having a generator in place not only for the meat and potatoes of Angular but also these arguably more tangential use cases is helpful as I can simply mention the generator syntax in our guidelines and documentation and take more comfort in code being generated uniformly across teams. This increased conformance then cuts out nuisance PR comments that may be the result of copy/paste code propagation.

Long story short, thank you! It is appreciated.
